File: frontend/src/dto/DashboardDTO.ts
import MoneyDTO from "./MoneyDTO";
import MonthDataDTO from "./MonthDataDTO";

export default interface DashboardDTO {
    budget: MoneyDTO,
    monthIncome: MoneyDTO,
    monthExpense: MoneyDTO,
    hhRate: MoneyDTO,
    savingsTotal: MoneyDTO,
    savingsLeft: MoneyDTO

    lastMonth: MonthDataDTO,
    currentMonth: MonthDataDTO,
    nextMonth: MonthDataDTO,
}

File: mft/src/main/java/com/github/arlekinside/diploma/ws/dto/DashboardDTO.java
package com.github.arlekinside.diploma.ws.dto;

import com.github.arlekinside.diploma.data.entity.Money;
import lombok.Data;

@Data
public class DashboardDTO {

    private Money budget = new Money();
    private Money monthIncome = new Money();
    private Money monthExpense = new Money();
    private Money hhRate = new Money();
    private Money savingsTotal = new Money();
    private Money savingsLeft = new Money();

    private MonthDataDTO lastMonth = new MonthDataDTO();
    private MonthDataDTO currentMonth = new MonthDataDTO();
    private MonthDataDTO nextMonth = new MonthDataDTO();

}
